tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -127,4 +131,35 @@ class PublicAssetSetCreatePerformer @Inject()(val publicAssetRepository: PublicA
         case _: BlobNotFoundException => PublicAssetBlobIdNotFoundException(creationRequest.blobId.value.value)
         case _: IllegalArgumentException => PublicAssetInvalidBlobIdException(creationRequest.blobId.value.value)
       }
+}
+
+
+class PublicAssetSetUpdatePerformer @Inject()(val publicAssetRepository: PublicAssetRepository,
+                                              val publicAssetSetService: PublicAssetSetService) {
+
+  def updatePublicAssets(mailboxSession: MailboxSession, request: PublicAssetSetRequest): SMono[PublicAssetUpdateResults] =
+    tryGetValidateUpdateRequest(request)
+      .flatMap({
+        case Left(updateFailure) => SMono.just(updateFailure)
+        case Right((publicAssetId, validatedPath)) => updatePublicAsset(publicAssetId, validatedPath, mailboxSession)
+      }, ReactorUtils.DEFAULT_CONCURRENCY)
+      .collectSeq()
+      .map(PublicAssetUpdateResults)
+
+  private def updatePublicAsset(publicAssetId: PublicAssetId, patch: ValidatedPublicAssetPatchObject, mailboxSession: MailboxSession): SMono[PublicAssetUpdateResult] =
+    publicAssetSetService.checkIdentityIdsExist(patch.identityIds.toSeq, mailboxSession)
+      .flatMap(checkedIdentityIds => SMono(publicAssetRepository.update(mailboxSession.getUser, publicAssetId, checkedIdentityIds.toSet)))
+      .`then`(SMono.just[PublicAssetUpdateResult](PublicAssetUpdateSuccess(publicAssetId)))
+      .onErrorResume(e => SMono.just(PublicAssetUpdateFailure(UnparsedPublicAssetId(publicAssetId.value.toString), e)))
```

Comment:
Please make sure that the failure is eventually logged.

Reply:
I did it in `PublicAssetUpdateFailure.asSetError` method
```
 def asSetError: SetError = exception match {
   ....
    case _ =>
      LOGGER.warn("Unexpected exception when update public asset ", exception)
      SetError.serverFail(SetErrorDescription(exception.getMessage))
  }
  
  ```